src/DomainHierarchy.ts

```diff
@@ -20,21 +21,21 @@ export class DomainHierarchy {
 
   constructor({
     domainReader,
-    ensRegistry,
+    ensRegistryAddress,
     provider,
     domainNotifierAddress,
     publicResolverAddress,
   }: {
     domainReader: DomainReader,
-    ensRegistry: ENSRegistry,
+    ensRegistryAddress: string,
     provider: providers.Provider,
     domainNotifierAddress: string
     publicResolverAddress?: string,
   }) {
     if (!domainReader) throw new Error("You need to pass a DomainReader");
     this._domainReader = domainReader;
-    if (!ensRegistry) throw new Error("You need to pass an ensRegistry ethers contract");
-    this._ensRegistry = ensRegistry;
+    if (!ensRegistryAddress) throw new Error("You need to pass an ensRegistry ethers contract");
```

Comment:
Maybe we should consider updating the error message

Reply:
Thanks, will do